packages/core/src/App/Components/Layout/Header/toggle-menu-drawer.jsx

```diff
@@ -352,40 +211,15 @@ const ToggleMenuDrawer = observer(({ platform_config }) => {
         );
     };
 
-    const getLanguageRoutes = () => {
+    const MobileLanguageMenu = React.useCallback(() => {
```

Comment:
useCallback is primarily used to memoize functions, not component. 
React provides React.memo specifically for this purpose 

Ref: https://react.dev/reference/react/memo

Reply:
yes, but in this case we need to move this component outside parent component and pass props.
we can’t use React.memo inside components.

Reply:
We can't use memo inside the component however we can't use this implementation as well as it won't produce the expected results. 
Either we can move the component outside or we can combine useMemo with useCallback to achieve the same behaviour as memo

Hope this helps: https://codesandbox.io/s/uplifted-state-with-memo-cjl2ht?file=/src/App.js 
But still moving the component outside with memo is a suggested approach

Reply:
will move it outside and pass the props accordingly